zcash_test_vectors/transaction.py

```diff
@@ -563,3 +575,151 @@ def __getattr__(self, item):
 
     def __bytes__(self):
         return bytes(self.inner)
+
+
+def legacy_test_vectors():
+    from hashlib import sha256
+
+    from .output import render_args, render_tv
+    from .rand import Rand
+    from .zip_0244 import txid_digest
+
+    args = render_args()
+
+    from random import Random
+    rng = Random(0xabad533d)
+    def randbytes(l):
+        ret = []
+        while len(ret) < l:
+            ret.append(rng.randrange(0, 256))
+        return bytes(ret)
+    rand = Rand(randbytes)
+
+    test_vectors = []
+    while len(test_vectors) < 30:
+        # Generate transactions with versions prior to ZIP 225.
+        tx = LegacyTransaction(rand, rand.u8() % NU5_TX_VERSION)
```

Comment:
0 was never a valid version number. I see that line 327 above will replace a version other than 3 (Overwinter) and 4 (Sapling) with a random version:
```
            self.nVersion = rand.u32() & ((1 << 31) - 1)
```
but I think that versions 1 and 2 (as version numbers that were actually used) should have high probability of occurrence, and 0 should be excluded.